### ISSUE TITLE
Add SAADC driver for nRF52

### ISSUE THREAD

**Issue by rodrigopex** [zephyrproject-rtos/zephyr#8643]
I have the SAADC working on my personal project using NRFX Hal, and I would like to "port" this code and make a driver for SAADC on Zephyr. Would I create a specific driver for SAADC cloning the ADC one or should I put the driver as another ADC driver implementation?

**Comment by carlescufi** [comment]
Hi @rodrigopex. Once this PR is merged you should have what you need:
https://github.com/zephyrproject-rtos/zephyr/pull/7691/commits

Mind closing this issue if that covers your requirement?

**Comment by rodrigopex** [comment]
Thank you. I will cover what I need.

**Comment by rodrigopex** [comment]
@carlescufi I have already done some code for this issue. As it is almost ready based on the #7691, I will discard my code. I did a sample for the ADC and I was wondering if it would be useful as we don't have any ADC sample code. This is the link for the branch with the implementation I did: https://github.com/rodrigopex/zephyr/tree/saadc_driver. It is really simple and basic but it solves what I need. It is using the "old" ADC API.
